storage/block_devices_ops.go

```diff
@@ -541,21 +545,593 @@ func partitionUsingParted(bd *BlockDevice, dryRun *[]string, wholeDisk bool) err
+		return err
+	}
+
+	raidParentFinder := func(b *BlockDevice) bool {
+		if b.FsType == "linux_raid_member" {
+			for _, ch := range b.Children {
+				if ch.Name == bd.Name {
+					return true
+				}
+			}
+		}
+		return false
+	}
+
+	parent, err := reverseLookParent(bd, disk, raidParentFinder)
+	if dryRun == nil {
+		log.Debug("removeRaidType: Found parent: %s for child: %s", parent.GetDeviceFile(), bd.GetDeviceFile())
+	}
+
+	if parent == nil {
+		return err
+	}
+
+	if dryRun == nil {
+		log.Warning("Strategy 1: Failing the RAID part: %s failed for RAID: %s", parent.GetDeviceFile(), bd.Name)
+		args := []string{"mdadm", "--fail", bd.GetDeviceFile(), parent.GetDeviceFile()}
+
+		err = cmd.RunAndLog(args...)
+		if err != nil {
+			goto strategy2
+		}
+
+		log.Warning("Removing the RAID part: %s from RAID: %s", bd.GetDeviceFile(), bd.Name)
+		args = []string{"mdadm", "--remove", bd.GetDeviceFile(), parent.GetDeviceFile()}
+
+		err = cmd.RunAndLog(args...)
+		if err != nil {
+			return errors.Wrap(err)
+		}
+
+		log.Warning("Zeroing RAID part super-block: %s", parent.GetDeviceFile())
+		args = []string{"mdadm", "--zero-superblock", parent.GetDeviceFile()}
+
+		err = cmd.RunAndLog(args...)
+		if err != nil {
+			return errors.Wrap(err)
+		}
+
+		return nil
+
+	strategy2:
+		log.Warning("Strategy 2: Stopping RAID: %s", bd.GetDeviceFile())
+		args = []string{"mdadm", "--stop", bd.GetDeviceFile()}
+
+		err = cmd.RunAndLog(args...)
+		if err != nil {
+			return errors.Wrap(err)
+		}
+
+		log.Warning("Zeroing RAID part super-block: %s", parent.GetDeviceFile())
+		args = []string{"mdadm", "--zero-superblock", parent.GetDeviceFile()}
+
+		err = cmd.RunAndLog(args...)
+		if err != nil {
+			return errors.Wrap(err)
+		}
+	} else {
+		*dryRun.CurrentTargetResults = append(*dryRun.CurrentTargetResults,
+			utils.Locale.Get("This will degrade RAID: %s", bd.Name))
+		*dryRun.CurrentTargetResults = append(*dryRun.CurrentTargetResults,
+			parent.Name+": "+utils.Locale.Get("Remove partition from RAID %s", bd.Name))
+	}
+
+	return nil
+}
+
+func removePart(bd *BlockDevice, ForceDestructive bool, disk string, dryRun *DryRunType, scans *preScanResults) error {
+	if bd.Type != BlockDeviceTypePart {
+		return errors.Errorf("Type is not a partition")
+	}
+
+	partNum := bd.GetPartitionNumber()
+
+	if partNum == 0 {
+		return errors.Errorf("Could not find partition number")
+	}
+
+	partParentFinder := func(b *BlockDevice) bool {
+		if b.Type == BlockDeviceTypeDisk || b.Type == BlockDeviceTypeLoop || b.isRaidType() {
+			for _, ch := range b.Children {
+				if ch.Name == bd.Name {
+					return true
+				}
+			}
+		}
+		return false
+	}
+
+	parent, err := reverseLookParent(bd, disk, partParentFinder)
+	if dryRun == nil {
+		log.Debug("removePart: Found parent: %s for child: %s", parent.GetDeviceFile(), bd.GetDeviceFile())
+	}
+
+	if parent == nil {
+		return err
+	}
+
+	if bd.FsType == "LVM2_member" {
```

Comment:
LVM2_member might be a good string to make into a const for reuse.

Reply:
make sense done